src/main/resources/SchemaForGrammars.xsd

```diff
@@ -277,7 +279,8 @@
                     <xs:complexType>
                         <xs:sequence>
                             <!-- datatype -->
-                            <xs:element name="charactersDatatypeID" type="xs:unsignedInt"/>
+                            <xs:element name="charactersDatatypeID" type="xs:unsignedInt"
+                                minOccurs="0"/>

```

Comment:
minOccurs="0" also not needed.

Reply:
May I ask you to change the schema just once (in the grammar section) given that this is the only place it should occur...

Thanks!